.github/workflows/check.yaml

```diff
@@ -8,21 +8,11 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 20
     steps:
-      - uses: actions/setup-go@v3
-        with:
-          go-version: '1.21'
       - name: Checkout code
-        uses: actions/checkout@v3
-      - name: Restore cache
-        uses: actions/cache@v3
+        uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
```

Comment:
change the `checkout` and `setup` place because https://github.com/actions/setup-node/issues/624#issuecomment-1515071902